### ISSUE TITLE
Task view: Keyboard navigation has many issues

### ISSUE THREAD

**Issue by AttemptToCallNil** [MediaWikiAGE/MAGE#59]
Filing this so that I don't forget and others know. At the moment, the features implemented for the task view work not that badly with mouse navigation, but keyboard navigation (Tab key switching) is very bad and allows bypassing modal restrictions, including pressing buttons that should be covered by the modal, and opening multiple modals of the same "level" simultaneously. Additionally, the generator / task type lists use heavily restyled radio buttons wrapped in `div`s (as I couldn't pick a proper semantic equivalent to what they do) and don't work with the keyboard just about at all.
